kubernetes/schema_container.go

```diff
@@ -211,7 +211,7 @@ func containerFields(isUpdatable, isInitContainer bool) map[string]*schema.Schem
 		"env": {
 			Type:        schema.TypeList,
 			Optional:    true,
-			ForceNew:    true,
+			ForceNew:    !isUpdatable,
```

Comment:
Looks like this line should just be removed as it is covered by https://github.com/terraform-providers/terraform-provider-kubernetes/blob/97f7696e7fc3e6e93b36706f85525217ff19ff15/kubernetes/schema_container.go#L539

Reply:
It looks to me like the entire container schema needs a second look at what's updateable and what's not.

Unfortunately, the Kubernetes client doesn't reflect that programmatically and the only way to determine is to run tests against an actual API and look at the validation responses.

Reply:
Agreed in the context of pods, but for pod templates, all fields are updatable and it is the job of the workload controller to ensure the pod are replaced.

Here, I essentially meant that the below _for loop_ already updates `ForceNew` to `True` on top level fields if `isUpdatable` is `False`:

```
	if !isUpdatable {
		for k := range s {
			if k == "image" && !isInitContainer {
				// this field is updatable for non-init containers
				continue
			}
			s[k].ForceNew = true
		}
	}
```

Reply:
Ah, that looks like a better way of doing it. I'll amend to remove the line in question.